restore status tooltip

brad-decker opened the pull request:
initially set out to add the failed tooltip back to the transaction list, but
in the process rediscovered the transaction-status component which illuminated
a fair number of statuses that were not properly handled by the refactor of the
list. These statuses were discussed with UX and engineering team members to come
up with a definitive list of statuses that should be reflected in the UI

Changes:
1. normalized the color of status labels to use Red-500 and Orange-500 where applicable
2. added a new color of icon for pending transactions -- grey
3. added support for dropped and rejected labels
4. failed, dropped, rejected and cancelled all have red icons now.
5. cancelled transactions will reflect a change in the user's balance
6. tooltip displayed for failed transactions
7. Icon logic isolated to a new component.

fixes #8731

Gudahtt commented:
>prior to this the queued "status" was only used inside the component, now it's being passed to the TransactionStatus component. I could either create a constant for this string or provide isEarliestNonce and isPending as additional props to TransactionStatus

That suggestion sounds reasonable. Or a prop like `isQueued` or something.

>at present I am reading into the error object on primaryTransaction inside TransactionListItem and using it to generate the tooltip. Would it make more sense to pass the err object into TransactionStatus and let it own the tooltip text generation?

I'm not sure :thinking: I'm having trouble having a preference here.

>CONFIRMED_STATUS is not really being used. In TransactionListItem we change the color of the icon and the transaction amount colors to indicate a failed transaction, but coincidently we do not change colors of these icons for canceled, dropped, etc. Should we?

`CONFIRMED_STATUS` does seem to be used, to determine whether the date is shown. The date implying confirmation seems OK I think :man_shrugging: 

It does seem appropriate to change the transaction amount colour for dropped as well - anything that indicates the transaction didn't succeed. I was interpreting the light-grey colour as indication that the transaction didn't effect the total.

Cancelled is a bit different in that it does effect the total, because the cancellation costs gas. So it did "succeed" in a sense. Not sure what to do there :thinking: 

>A whole host of statuses that were previously labels are not used anymore. Should we still apply a class modifier to these or remove them from the classHash in TransactionStatus

We should definitely make some decision about how to display them. Maybe they don't need their own classes though - that's less important than ensuring they show up how we want them to.

e.g. which statuses should be shown at all, which should be red v.s. blue, etc.

brad-decker commented on ui/app/components/app/transaction-status/transaction-status.component.js:
```diff
@@ -15,52 +14,46 @@ import {
+      >
+        {statusText }
+      </Tooltip>
+      {' '}·{' '}
```
yeah it should just be one string literal with spacing around that little dot. was trying to enforce spacing to stay in place but it makes more sense as `{' · '}`

brad-decker committed:
update transaction status UI

initially set out to add the failed tooltip back to the transaction list, but
in the process rediscovered the transaction-status component which illuminated
a fair number of statuses that were not properly handled by the refactor of the
list. These statuses were discussed with UX and engineering team members to come
up with a definitive list of statuses that should be reflected in the UI

Changes:
1. normalized the color of status labels to use Red-500 and Orange-500 where applicable
2. added a new color of icon for pending transactions -- grey
3. added support for dropped and rejected labels
4. failed, dropped, rejected and cancelled all have red icons now.
5. cancelled transactions will reflect a change in the user's balance
6. tooltip displayed for failed transactions
7. Icon logic isolated to a new component.

Gudahtt reviewed (approved):
LGTM!
